frontend/src/components/page_content/plan/debt.js

```diff
@@ -24,31 +26,55 @@ class Debt extends React.Component {
         fill: "var(--toast-neutral-3)",
       },
       {
-        name: "Target Savings",
+        name: "Max Savings",
         value: this.props.debtMonthly,
         fill: "url(#gradient)",
       },
     ];
+
+    this.actual = [
+      {
+        name: "Salary After Debt",
+        value:
+          calcMonthlyValue(this.props.salaryAfterTax) - this.props.loanDebt ||
+          0,
+        fill: "var(--toast-neutral-3)",
+      },
+      {
+        name: "Current Savings",
+        value: parseFloat(this.props.loanDebt || "0"),
+        fill: "url(#gradient)",
+      },
+    ];
+
+    this.setActiveToggle = this.setActiveToggle.bind(this);
+    this.getOnTrack = this.getOnTrack.bind(this);
+
+    this.currentDebt = React.createRef();
+    this.targetDebt = React.createRef();
   }
 
   getHeader() {
     return `$${numWithCommas(this.props.debtMonthly)}`;
   }
 
   getOnTrack() {
-    if (this.props.isOnTrackDebt === undefined) {
+    //let isOntrack = this.props.isOnTrackDebt; // Use backend
+    let isOntrack =
+      parseFloat(this.props.loanDebt || "0") <= this.props.debtMonthly; // Use calculated
```

Comment:
you shouldnt need parse float because it should have been turned into a number in effects/index.js when the value is read in